### PR TITLE
Check that generation of MooseRevision.h is successful before continuing compilation.

### PR DESCRIPTION
**Pull request by pshriwise**
This PR is related to https://github.com/idaholab/moose/discussions/22035. The generation of the `MooseRevision.h` header relies on the Python `packaging` module. If this package is missing, an error message appears at the start of compilation:

```shell
 Traceback (most recent call last):
  File "/home/pshriwise/external/cardinal-aurora/build/moose/framework/scripts/get_repo_revision.py", line 19, in <module>
    from packaging import version
ModuleNotFoundError: No module named 'packaging'
```

As it stands, compilation will continue until getting to a file that includes that header with a message like:

```shell
In file included from /home/pshriwise/external/cardinal-aurora/build/moose/framework/build/unity_src/outputs_Unity.C:16:
/home/pshriwise/external/cardinal-aurora/build/moose/framework/src/outputs/JsonIO.C:12:10: fatal error: MooseRevision.h: No such file or directory
   12 | #include "MooseRevision.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```

The fact that the final error message comes from the compiler but is related to a missing Python package might lead some down the wrong path when they look into fixing this problem.

## Reason
Makes the source of a compilation error related to a missing Python package clearer.

## Design
Added a check on the returned error code from the Python command in the header generation target.

## Impact
Hopefully will make the source of the failure clearer and easier to fix.

### CODE REVIEW

**Comment by GiudGiud** [comment]
The commit messages will need to refer to an actual issue (not a discussions)
#22090 maybe?
or a new one

**Comment by pshriwise** [comment]
> The commit messages will need to refer to an actual issue (not a discussions)
> #22090 maybe?
> or a new one

Can do! 

I created a new one and connected the issue you mentioned above as well as the discussion. I then added a reference to the new issue in the last commit message. Let me know if you'd like that done differently.

File: framework/moose.mk [hunk 1]
```diff
@@ -317,6 +317,10 @@ $(moose_revision_header): $(moose_HEADER_deps)
 	@echo "Checking if header needs updating: "$@"..."
 	$(shell $(FRAMEWORK_DIR)/scripts/get_repo_revision.py $(FRAMEWORK_DIR) \
 	  $(moose_revision_header) MOOSE)
+        # make sure the header generation step didn't fail
+	@if [ $(.SHELLSTATUS) -ne 0 ]; then \
+	echo "\nFailed to generate MooseRevision.h\n"; exit $(.SHELLSTATUS); \
+	fi
 	@if [ ! -e "$(moose_all_header_dir)/MooseRevision.h" ]; then \
 		ln -sf $(moose_revision_header) $(moose_all_header_dir); \
```
Comment (GiudGiud) [inline framework/moose.mk:325]: @loganharbour this code makes me think that we used to be fine with not having this header?

**Comment by pshriwise** [comment]
Anything I can do here to move things along?

File: framework/moose.mk [hunk 2]
```diff
@@ -317,6 +317,10 @@ $(moose_revision_header): $(moose_HEADER_deps)
 	@echo "Checking if header needs updating: "$@"..."
 	$(shell $(FRAMEWORK_DIR)/scripts/get_repo_revision.py $(FRAMEWORK_DIR) \
 	  $(moose_revision_header) MOOSE)
+        # make sure the header generation step didn't fail
+	@if [ $(.SHELLSTATUS) -ne 0 ]; then \
```
Comment (GiudGiud) [inline framework/moose.mk:321]: we're good on forcing GNU Make >=4.2 ?
@loganharbour

File: framework/moose.mk [hunk 3]
```diff
@@ -317,6 +317,10 @@ $(moose_revision_header): $(moose_HEADER_deps)
 	@echo "Checking if header needs updating: "$@"..."
 	$(shell $(FRAMEWORK_DIR)/scripts/get_repo_revision.py $(FRAMEWORK_DIR) \
 	  $(moose_revision_header) MOOSE)
+        # make sure the header generation step didn't fail
```
Comment (GiudGiud) [inline framework/moose.mk:320]:
```suggestion
  # make sure the header generation step didn't fail
```

File: framework/moose.mk (hunk 2 above)
Comment (loganharbour) [reply framework/moose.mk:321]: Hmm... I think this is fine. Casually requiring six year old make.

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the fix. The way it errors as of now is much better than before 
I m thinking in the future we could check for these python packages a little stronger as well since it's the real problem.

**Pull request merged**